### PR TITLE
Generic iterator token

### PR DESCRIPTION
**Pull request by Easyoakland**
Generic token that applies to all clone-able iterators.
For example works to parse `Iterator<Item = u8>`

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -174,6 +174,77 @@ impl<'a> IntoTokens<char> for &'a str {
...
+where
+    I: Iterator + Clone,
+{
+    type Item = I::Item;
+
+    type Location = IteratorTokensLocation<I>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.cursor += 1;
+        self.iter.next()
```
Comment (jsdw) [inline src/types.rs:219]: If `self.iter.next()` returns `None`, we probably shouldn't update the cursor :)

**Comment by jsdw** [comment]
Thanks for the PR!

Let me think about this one; one reason I've not added this yet is that I've been thinking about how to return slices of tokens back more ergonomically than they are at the mo (eg with a couple of `StrTokens::location()`s it'd be great to be able to get an `&str` back, and likewise with `SliceTokens` and `&[u8]`. If this is possible, how it works will probably have an impact on how to work with iterators (though I would like to preserve this ability too and be able to hand back a sub-iterator or something!)

File: src/types.rs (hunk 1 above)
Comment (Easyoakland) [reply src/types.rs:219]: - If the iterator is not `fused` then a `None` does not indicate that something hasn't happened to the iterator's internal state. For example, a non-fused iterator could return None before returning elements again.
- I tried to match the meaning of .nth(offset). If the iterator would only yield 2 items .nth(100) would still call next 100 times.

Comment (jsdw) [reply src/types.rs:219]: Those seem like fair points; I can probably get behind that!

**Comment by jsdw** [comment]
Getting this merged in #10; thankyou for your contribution :)

**Pull request closed without merging**